plugins/civo/api_key.go

```diff
@@ -0,0 +1,138 @@
+package civo
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	//"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://www.civo.com/docs/account/api-keys"),
+		ManagementURL: sdk.URL("https://dashboard.civo.com/security"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Civo.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 50,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "The Name of apikey used to authenticate to civo",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: civoProvisioner{},
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			importer.TryEnvVarPair(secondEnvVarMapping),
+			TryCivoConfigFile("~/.civo.json"),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"CIVO_TOKEN": fieldname.APIKey,
+}
+
+var secondEnvVarMapping = map[string]sdk.FieldName{
+	"CIVO_API_KEY":      fieldname.APIKey,
+	"CIVO_API_KEY_NAME": fieldname.APIKeyID,
+}
+
+type civoProvisioner struct {
+	sdk.Provisioner
+	Schema map[string]sdk.FieldName
+}
+
+func (p civoProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	if apiKey, ok := in.ItemFields[fieldname.APIKey]; ok {
+		out.AddEnvVar("CIVO_TOKEN", apiKey)
+	}
+
+	if apiKey, ok := in.ItemFields[fieldname.APIKey]; ok {
+		out.AddEnvVar("CIVO_API_KEY", apiKey)
+		//out.AddEnvVar("CIVO_API_KEY_NAME", apiKeyID)
+	}
```

Comment:
With the understanding that `CIVO_API_KEY` and `CIVO_TOKEN` hold the same API Key, why are we provisioning the same secret twice?

Reply:
@arunsathiya I will update it.